Feature/log less

eventengineering opened the pull request:
Answers #298

Small modifications to update log lines to imply without confusion where the log lines are coming from.

logLevel property on all Android and iOS sensor objects.

setLogLevel function on all Android and iOS sensor objects.

Use of NSLog on iOS objects put behind a logLevel check where applicable.

eventengineering commented:
These responses from commitlint do not make sense and produces an unnecessary barrier to having anyone work on the code (a barrier to the biggest benefit of open source projects).

eventengineering commented:
The Travis configuration seems rather broken.

DanielMSchmidt commented:
Hey @eventengineering 

> These responses from commitlint do not make sense and produces an unnecessary barrier to having anyone work on the code (a barrier to the biggest benefit of open source projects).

I'm sorry I need to enforce us using semantic commit messages, so that the automatic release process works properly :) 

> The Travis configuration seems rather broken.

No, the test is a legit fail, because your code is not formatted via prettier. A quick `npm run format` should fix the issue 👍 I'll also add a pre-commit hook so that the commit experience is better

eventengineering commented:
There seems to be a mis-configuration with travis-ci: "no environment variables set"

eventengineering commented:
That's not what travis-ci reports. "no environment variables set"

eventengineering commented:
Asking someone to run `npm run format` might not be the best thing for PRs on this project: there's tonnes of files that are not correctly formatted and get changed as a result.

You've also got something imported that tries to delete npm from the node_modules folder, despite it not existing. I had to create an empty directory called npm in my node_modules folder for `npm install` to function correctly on the repo.

DanielMSchmidt commented:
@eventengineering Thank you for raising this, I wasn't aware that there are changes on master due to this. I'll fix it after merging this 👍

DanielMSchmidt commented:
:tada: This PR is included in version 5.4.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-native-sensors)
- [GitHub release](https://github.com/react-native-sensors/react-native-sensors/releases/tag/v5.4.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

DanielMSchmidt commented:
Okay, I checked, there are actually no files that need reformatting, its just that prettier logs out every file it looks at and if it's changed it highlights it and otherwise the file names are printed darker. I changed the settings so the output is empty if no files change to prevent future confusion.

eventengineering commented:
Huh, that's strange. I had changes in over 20 files, including tests, documentation, and code in files that I had not edited just from running this.

DanielMSchmidt commented:
Did these changes appear also in `git diff`?

eventengineering commented:
After running `npm run format` I get the following changes. All changes to whitespace.
```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   examples/BasicSensors/App.js
        modified:   examples/BasicSensors/SensorView.js
        modified:   examples/BasicSensors/ios/BasicSensors/Images.xcassets/AppIcon.appiconset/Contents.json
        modified:   examples/BasicSensors/ios/BasicSensors/Images.xcassets/Contents.json
        modified:   examples/GryoscopeImage/App.js
        modified:   examples/GryoscopeImage/ios/GryoscopeImage/Images.xcassets/AppIcon.appiconset/Contents.json
        modified:   examples/GryoscopeImage/ios/GryoscopeImage/Images.xcassets/Contents.json
        modified:   examples/KeepTheBallGame/App.js
        modified:   examples/KeepTheBallGame/Ball.js
        modified:   examples/KeepTheBallGame/Game.js
        modified:   examples/KeepTheBallGame/GameOver.js
        modified:   examples/KeepTheBallGame/Table.js
        modified:   examples/KeepTheBallGame/ios/KeepTheBallGame/Images.xcassets/AppIcon.appiconset/Contents.json
        modified:   examples/KeepTheBallGame/ios/KeepTheBallGame/Images.xcassets/Contents.json
        modified:   src/rnsensors.js
        modified:   src/sensors.js
        modified:   tests/App.js
        modified:   tests/SensorView.js
        modified:   tests/e2e/accelerometer.spec.js
        modified:   tests/e2e/config.json
        modified:   tests/ios/tests/Images.xcassets/AppIcon.appiconset/Contents.json
        modified:   tests/ios/tests/Images.xcassets/Contents.json
        modified:   website/core/Footer.js
        modified:   website/i18n/en.json
        modified:   website/pages/en/index.js
        modified:   website/sidebars.json
        modified:   website/siteConfig.js

no changes added to commit (use "git add" and/or "git commit -a")
```

eventengineering commented:
Ah, I have a funny feeling it's the formatter conflicting with my git smudge/clean. My apologies.

DanielMSchmidt commented:
Ah okay, that could be. Is this a setup that is special to you or do you think we should add a note to the contributing docs?

eventengineering commented:
Others are just as likely to encounter this, but should be aware of the hiccups it might cause with any formatter.

Such will always be an issue with space indentation.

As for the other issue, the `npm` folder removal: I've not found the dependency responsible, but it hasn't happened since, even if I clear the whole npm folder.

jdegger commented on ios/Barometer.m:
```diff
@@ -56,30 +57,47 @@ - (void) isAvailableWithResolver:(RCTPromiseResolveBlock) resolve
+
         if (altitudeData) {
+            if (self->logLevel > 1) {
+                NSLog(@"Updated altitue value: %f, %f", altitudeData.pressure.doubleValue, timestamp);
```
Timestamp is missing here, since this PR our app is not building anymore. Undeclared identifier timestamp.
